### PR TITLE
Generate separated image with manifests

### PR DESCRIPTION
**Pull request by mhmxs**
Operator manifests now generated as a separated docker image for both develop and tagged versions.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ env:
     - K8S_VERSION="v1.19.0"
     - MINIKUBE_VERSION="v0.28.2"
     - IMAGE_NAME=storageos/cluster-operator
-    - IMAGE_TAG=test
```
Comment (mhmxs) [inline .travis.yml:14]: Previously for `operator-image` target we used `IMAGE_TAG` but for `install-manifest` we used `TRAVIS_TAG` and then we re-tagged image. With this change we build only `TRAVIS_TAG` so w save an extra tagging.

File: .travis.yml [hunk 2]
```diff
@@ -71,8 +71,8 @@ jobs:
             echo "Docker credentials not set. Skipping docker log in...";
           fi
       script:
-        - make operator-image
-        - OPERATOR_IMAGE=storageos/cluster-operator:$TRAVIS_TAG make install-manifest
+        - |
+          (: ${TRAVIS_TAG:=develop} ; OPERATOR_IMAGE=$IMAGE_NAME:$TRAVIS_TAG MANIFESTS_IMG=$MANIFESTS_IMAGE_NAME:cluster-$TRAVIS_TAG make operator-image install-manifest manifests-image)
```
Comment (mhmxs) [inline .travis.yml:75]: Set `TRAVIS_TAG` if not setted.

File: Makefile [hunk 3]
```diff
@@ -1,4 +1,6 @@
 OPERATOR_IMAGE ?= storageos/cluster-operator:test
+# Image URL for manifest image.
+MANIFESTS_IMG ?= storageos/operator-manifests:cluster-test
```
Comment (croomes) [inline Makefile:3]: We use `test` in most places - any reason for `cluster-test`?

File: scripts/deploy.sh [hunk 4]
```diff
@@ -31,6 +32,6 @@ if [ "$1" = "tagged" ]; then
        tianon/github-hub:2 bash -c "./scripts/create-pr.sh"
 
 elif [ "$1" = "develop" ]; then
-    docker tag "${IMAGE_NAME}:${IMAGE_TAG}" "${IMAGE_NAME}:develop"
     docker push "${IMAGE_NAME}:develop"
+    docker push "${MANIFESTS_IMAGE_NAME}:cluster-develop"
```
Comment (croomes) [inline scripts/deploy.sh:36]: `develop`?

**Review by croomes — COMMENTED** [review]
A couple minor naming things that might be fine, they just appear inconsistent.

File: Makefile (hunk 3 above)
Comment (mhmxs) [reply Makefile:3]: New operator uses the same docker repo (as you suggested), and we are not able to use the same tag.

File: scripts/deploy.sh (hunk 4 above)
Comment (mhmxs) [reply scripts/deploy.sh:36]: https://github.com/storageos/cluster-operator/pull/350#discussion_r703644066

File: Makefile (hunk 3 above)
Comment (mhmxs) [reply Makefile:3]: If you create a separated repo for cluster operator manifests, i'm glad to use it.

Comment (croomes) [reply Makefile:3]: makes sense - thanks

**Review by croomes — APPROVED** [review]

**Pull request merged**